example/conduit/t8forest_to_conduit_mesh.cxx

```diff
@@ -0,0 +1,429 @@
+                          int write_element_id, int curved_flag,
+                          int num_data, t8_vtk_data_field_t *data)
+{
+
+  /*
+   * Nutzen diesmal data direkt statt array of array struct dafür zu verwenden.
+   * Currently curved elements and prisms and pyramides are not supported
+   * but can be added by using polygonal/polyhedral elements
+   */
+
+  /* Check assertions: forest and fileprefix are not NULL and forest is commited */
+  T8_ASSERT (forest != NULL);
+  T8_ASSERT (forest->rc.refcount > 0);
+  T8_ASSERT (forest->committed);
+
+  long int            point_id = 0;     /* The id of the point in the points Object. */
+  t8_locidx_t         ielement; /* The iterator over elements in a tree. */
+  t8_locidx_t         itree, ivertex;
+  double              coordinates[3];
+  double              vertex_coords[3] = { 0, 0, 0 };
+  int                 elem_id = 0;
+  t8_locidx_t         num_elements;
+  t8_gloidx_t         gtreeid;
+  t8_cmesh_t          cmesh;
+  int                 num_corners;
+
+  /* Here we construct the mesh node object, s.t. it is an explicit, unstructured mesh
+   * with mixed element types. The shape map contains the corresponding element int numbers
+   * for every element type. 
+   */
+
+  Node                mesh;
+  mesh["coordsets/coords/type"] = "explicit";
+  mesh["topologies/mesh/type"] = "unstructured";
+  mesh["topologies/mesh/coordset"] = "coords";
+  mesh["topologies/mesh/elements/shape"] = "mixed";
+  mesh["topologies/mesh/elements/shape_map/point"] = 1;
+  mesh["topologies/mesh/elements/shape_map/line"] = 3;
+  mesh["topologies/mesh/elements/shape_map/tri"] = 5;
+  mesh["topologies/mesh/elements/shape_map/quad"] = 9;
+  mesh["topologies/mesh/elements/shape_map/tet"] = 10;
+  mesh["topologies/mesh/elements/shape_map/hex"] = 12;
+
+  /* 
+   * Furthermore, we need
+   * under topologies/topo/elements/shapes : (shapes array), which
+   * contains the int number of each element in the mesh,
+   * under topologies/topo/elements/sizes : (sizes array), which
+   * contains the sizes of those elements,
+   * topologies/topo/elements/offsets : (offsets array), which
+   * contains the offset, meaning the index at which each element starts in the points array
+   * and topologies/topo/elements/connectivity : (connectivity array), which
+   * contains the points ids that make up an element for each element. 
+   * We need to allocate a num points long array for all but the connectivity array.
+   * Thus we first calculate the number of points.
+   */
+
+  num_elements = t8_forest_get_local_num_elements (forest);
+  int                 connec_iter = 0;
+
+  /* 
+   * Additionally, for the connectivity array we need to allocate the sum
+   * of the number of points over all elements in the mesh. We store this in 
+   * the connec_iter. May need a better name. 
+   */
+
+  for (itree = 0; itree < t8_forest_get_num_local_trees (forest); itree++) {
```

Comment:
```suggestion
  for (t8_locidxt_t itree = 0; itree < t8_forest_get_num_local_trees (forest); itree++) {
```
Our coding-guidelines regarding variable-declaration has been updated recently. Please have a look at it: https://github.com/DLR-AMR/t8code/wiki/Coding-Guideline